### PR TITLE
Use `MPItrampoline` as ABI name

### PR DESCRIPTION
**Pull request by eschnett**
See https://github.com/JuliaPackaging/Yggdrasil/pull/4750. Both patches need to be applied simultaneously.

### CODE REVIEW

File: lib/MPIPreferences/src/MPIPreferences.jl [hunk 1]
```diff
@@ -149,8 +149,8 @@ function identify_abi(libmpi)
         if (m = match(r"^FUJITSU MPI Library (\d+.\d+.\d+)", version_string)) !== nothing
             version = VersionNumber(m.captures[1])
         end
-    elseif startswith(version_string, "MPIwrapper")
-        impl = "MPIwrapper"
+    elseif startswith(version_string, "MPItrampoline")
+        impl = "MPItrampoline"
         # MPIwrapper 2.2.2
         if (m = match(r"^MPIwrapper Version:\t(\d+.\d+.\d+\w*)", version_string)) !== nothing
```
Comment (simonbyrne) [inline lib/MPIPreferences/src/MPIPreferences.jl:155]: Does this need to be changed as well? It should match line 152.

**Comment by giordano** [comment]
Duplicate of #549?

File: lib/MPIPreferences/src/MPIPreferences.jl [hunk 2]
```diff
@@ -149,8 +149,8 @@ function identify_abi(libmpi)
         if (m = match(r"^FUJITSU MPI Library (\d+.\d+.\d+)", version_string)) !== nothing
             version = VersionNumber(m.captures[1])
         end
-    elseif startswith(version_string, "MPIwrapper")
-        impl = "MPIwrapper"
+    elseif startswith(version_string, "MPItrampoline")
```
Comment (vchuravy) [inline lib/MPIPreferences/src/MPIPreferences.jl:152]: Is this change necessary? I think this is the only one I didn't have in my PR

Comment (eschnett) [reply lib/MPIPreferences/src/MPIPreferences.jl:152]: The `version_string` remains `MPIwrapper`.

**Comment by vchuravy** [comment]
Should all be on master :)
